cvat-ui/src/components/common/cvat-table.tsx

```diff
@@ -0,0 +1,274 @@
+// Copyright (C) CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import React, { useCallback, useEffect, useState } from 'react';
+import { Row, Col } from 'antd/lib/grid';
+import Table, { TableProps } from 'antd/lib/table';
+import Input from 'antd/lib/input';
+import Text from 'antd/lib/typography/Text';
+import Button from 'antd/lib/button';
+import Space from 'antd/lib/space';
+import Popover from 'antd/lib/popover';
+import Checkbox, { CheckboxChangeEvent } from 'antd/lib/checkbox';
+import { DownloadOutlined, MoreOutlined } from '@ant-design/icons';
+import { Config } from '@react-awesome-query-builder/antd';
+import jsonLogic from 'json-logic-js';
+
+import { ResourceFilterHOC, defaultVisibility } from 'components/resource-sorting-filtering';
+
+type Props = TableProps & {
+    onFilterDataSource?(data: TableProps['dataSource']): void;
+    onChangeColumnVisibility?(id: number, isHidden: boolean): void;
+    renderExtraActions?(): JSX.Element;
+    queryBuilderConfig?: Partial<Config>;
+    queryBuilderPredefinedQueries?: Record<string, string>;
+    queryBuilderCache?: {
```

Comment:
Maybe its better to be inside of a object `query: {buiderConfig, builderPredifinedQueries, etc }`